src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java

```diff
@@ -288,7 +297,17 @@ public UUID getStudyRootNodeUuid(UUID studyId) {
 
     @Transactional
     public Optional<UUID> doGetModificationGroupUuid(UUID nodeUuid, boolean generateId) {
-        return nodesRepository.findById(nodeUuid).flatMap(n -> repositories.get(n.getType()).getModificationGroupUuid(nodeUuid, generateId));
+        Optional<NodeEntity> nodeEntity = nodesRepository.findById(nodeUuid);
```

Comment:
exactly the same as doGetVariantId, except the last line, can use a (private) function

Reply:
Code factorized

Reply:
ok